src/CLU/clurun/Program.cs

```diff
@@ -15,16 +21,76 @@ public class Program
         /// </summary>
         /// <param name="args">The commandline arguments</param>
         public static int Main(string[] args)
-        { 
+        {
             var debugClu = Environment.GetEnvironmentVariable("DebugCLU");
             if (!String.IsNullOrEmpty(debugClu))
             {
                 System.Console.WriteLine("This is your chance to attach a debugger...");
                 System.Console.ReadLine();
             }
-            return (int) Microsoft.CLU.Run.CLURun.Execute(args);
 
-            
+            if (args.Contains("--install"))
+            {
+                return (int)Microsoft.CLU.Run.CLURun.Execute(args);
+            }
+
+            string argsString = string.Join(" ", args);
+            string[] indexFiles = Directory.GetFiles(GetExeDirectory(), "*.idx", SearchOption.AllDirectories);
+
+            var commands = indexFiles
+                .SelectMany(f => File.ReadAllLines(f))
+                .Select(f => new CommandIndex(f));
+
+            var command = commands
+                .OrderByDescending(c => 
+                    CommandMatcher.GetMatchScore(c.Args, argsString))
+                    .FirstOrDefault(c => argsString.Contains(c.Args));
+
+            if (command == null)
+            {
+                ShowHelp(argsString, commands);
+                return -1;
+            }
+
+            return Execute(command, args);
+        }
+
+        private static string GetExeDirectory()
+        {
+            var assemblyLocation = typeof(Program).GetTypeInfo().Assembly.Location;
+            return Path.GetDirectoryName(assemblyLocation);
+        }
+
+        private static int Execute(CommandIndex command, string[] args)
+        {
+            string executablePath = Directory.GetFiles(
+                GetExeDirectory(), 
+                command.Package + Platform.ExecutableExtension, 
+                SearchOption.AllDirectories).FirstOrDefault();
+
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+            startInfo.FileName = executablePath;
+            // Add extra quotes since we are passing on to another process
+            startInfo.Arguments = string.Join(" ", args.Select(a => $"\"{a.Replace("\"", "\"\"")}\""));
```

Comment:
Tested that args work with quotes and escaped quotes.  I wrap the arg in quotes to support quoting and also double the number of quotes inside the arg to support escaping